src/platform/plugins/shared/unified_doc_viewer/public/components/observability/attributes/doc_viewer_attributes_overview/attributes_overview.tsx

```diff
@@ -53,32 +72,50 @@ export function AttributesOverview({
 
   const allFields = Object.keys(flattened);
 
-  // it filters out multifields that have a parent, to prevent entries for multifields like this: field, field.keyword, field.whatever
-  const filteredFields = useMemo(
-    () => allFields.filter(shouldShowFieldHandler),
-    [allFields, shouldShowFieldHandler]
-  );
-
   const groupedFields = useMemo(() => {
-    const attributesFields: string[] = [];
-    const resourceAttributesFields: string[] = [];
-    const scopeAttributesFields: string[] = [];
+    const attributesFields: AttributeField[] = [];
+    const resourceAttributesFields: AttributeField[] = [];
+    const scopeAttributesFields: AttributeField[] = [];
     const lowerSearchTerm = searchTerm.toLowerCase();
-    for (const fieldName of filteredFields) {
+
+    allFields.reduce((acc, fieldName) => {
       const lowerFieldName = fieldName.toLowerCase();
+
+      // it filters out multifields that have a parent, to prevent entries for multifields like this: field, field.keyword, field.whatever
+      if (!shouldShowFieldHandler(fieldName)) {
+        return acc;
+      }
+
       if (!lowerFieldName.includes(lowerSearchTerm)) {
-        continue;
+        return acc;
+      }
+
+      if (isEsqlMode && areNullValuesHidden && flattened[fieldName] == null) {
+        return acc;
       }
+
       if (lowerFieldName.startsWith('resource.attributes.')) {
-        resourceAttributesFields.push(fieldName);
+        resourceAttributesFields.push({
```

Comment:
There’s some repetition in this approach. 
 Can we improve it a bit?  We could use a mapping something like or something similar 

````
const fieldTypeMap = [
  { prefix: 'resource.attributes.', target: resourceAttributesFields },
  { prefix: 'scope.attributes.', target: scopeAttributesFields },
  { prefix: 'attributes.', target: attributesFields },
];

for (const { prefix, target } of fieldTypeMap) {
  if (lowerFieldName.startsWith(prefix)) {
    target.push({
      name: fieldName,
      displayName: getAttributeDisplayName(fieldName),
    });
  }
}
```